### PR TITLE
Rename makefile, improve tooling

### PR DESCRIPTION
**Pull request by braydonk**
## Description
Added yamlfmt tool for yaml formatting, addlicense for license header
checks, and some helpful precommit make targets using the new tools.

## Related issue
b/238471196

## How has this been tested?
`make -f tasks.mak precommit`: https://pastebin.com/uxKg1XF3
`make -f tasks.mak addlicense`: The results are part of the changes in this PR
`make -f tasks.mak test_confgenerator_update`: The results are part of the changes in this PR

## Checklist:
- Unit tests
  - [ ] Unit tests do not apply.
  - [x] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [x] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

**Comment by braydonk** [comment]
`PR_Presubmit` is going to fail here because it thinks I changed the supported agent version, but it's only an indentation change so it's fine to ignore.

File: tasks.mak [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+test_confgenerator_update:
+	go test -mod=mod github.com/GoogleCloudPlatform/ops-agent/confgenerator -update_golden
+
+test_metadata:
+	go test ./integration_test/metadata
+
+test_metadata_update:
+	go test ./integration_test/metadata -update_golden
+
+validate_metadata:
```
Comment (qingling128) [inline tasks.mak:60]: [Optional] `s/validate_metadata/test_validate_metadata/g` for consistency?

File: tasks.mak [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Copyright 2022 Google LLC
```
Comment (qingling128) [inline tasks.mak:1]: Can we add some documentation of this tool to:
* https://github.com/GoogleCloudPlatform/ops-agent/tree/master/dev-docs (e.g. a new `dev-tool.md` file)
* Internal dev guide (can point to the new `.md` file above)

**Review by qingling128 — APPROVED** [review]
LGTM

**Comment by qingling128** [comment]
LGTM

**Pull request merged**